pkg/chunk/aws/s3_storage_client.go

```diff
@@ -100,7 +108,10 @@ func (cfg *S3Config) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	f.StringVar(&cfg.AccessKeyID, prefix+"s3.access-key-id", "", "AWS Access Key ID")
 	f.StringVar(&cfg.SecretAccessKey, prefix+"s3.secret-access-key", "", "AWS Secret Access Key")
 	f.BoolVar(&cfg.Insecure, prefix+"s3.insecure", false, "Disable https on s3 connection.")
-	f.BoolVar(&cfg.SSEEncryption, prefix+"s3.sse-encryption", false, "Enable AES256 AWS Server Side Encryption")
+	f.BoolVar(&cfg.SSEEncryption, prefix+"s3.sse-encryption", false, "Enable AWS Server Side Encryption [Deprecated: Use sse_config instead. if s3.sse-encryption is enabled, it assumes sse_config SSE-S3 type]")
+	f.StringVar(&cfg.SSEConfig.Type, prefix+"s3.sse_config.type", "", "Enable AWS Server Side Encryption. Only SSE-S3 and SSE-KMS are supported")
+	f.StringVar(&cfg.SSEConfig.KMSKeyID, prefix+"s3.sse_config.kms_key_id", "", "KMS Key ID used to encrypt objects in S3")
+	f.Var(&cfg.SSEConfig.KMSEncryptionContext, prefix+"s3.sse_config.kms_encryption_context", "KMS Encryption Context used for object encryption")
```

Comment:
CLI flags are `-` separated, while YAML config options are `_` separated.

```suggestion
	f.StringVar(&cfg.SSEConfig.KMSKeyID, prefix+"s3.sse_config.kms-key-id", "", "KMS Key ID used to encrypt objects in S3")
	f.Var(&cfg.SSEConfig.KMSEncryptionContext, prefix+"s3.sse_config.kms-encryption-context", "KMS Encryption Context used for object encryption")
```